test/lib/release_notes_reader_test.rb

```diff
@@ -2,160 +2,165 @@
-      dir = Dir.mktmpdir
-      allow(Dir).to receive(:mktmpdir).and_return(dir)
-      reader.release_notes_for(product)
-      expect(File).to_not be_directory(dir)
-    end
+  describe "#release_notes" do
+    let(:registered?) { true }
 
-    it "stores the result for later retrieval" do
-      expect(release_notes_store).to receive(:store)
-        .with(Y2Packager::ReleaseNotes)
-      reader.release_notes_for(product)
+    before do
+      stub_const("Registration::Registration", registration)
+      allow(reader).to receive(:require).with("registration/registration")
     end
 
-    context "when the release notes were already downloaded" do
-      let(:relnotes) { instance_double(Y2Packager::ReleaseNotes) }
+    context "when system is registered" do
+      let(:registered) { true }
 
-      before do
-        allow(release_notes_store).to receive(:retrieve)
-          .and_return(relnotes)
+      it "retrieves release notes from RPM packages" do
+        expect(rpm_reader).to receive(:release_notes).with(
+          Y2Packager::ReleaseNotesContentPrefs.new("de_DE", "en", :txt)
+        )
+        rn = reader.release_notes(user_lang: "de_DE", format: :txt)
+        expect(rn).to eq(release_notes)
       end
 
-      it "does not download them again" do
-        expect(reader.release_notes_for(product)).to eq(relnotes)
-      end
+      context "when release notes are not found" do
+        let(:release_notes) { nil }
 
-      it "does not try to store the result" do
-        expect(release_notes_store).to_not receive(:store)
-        expect(reader.release_notes_for(product)).to eq(relnotes)
+        it "tries to get release notes from the relnotes_url property" do
+          expect(rpm_reader).to receive(:release_notes).with(
+            Y2Packager::ReleaseNotesContentPrefs.new("de_DE", "en", :html)
+          )
+          rn = reader.release_notes(user_lang: "de_DE", format: :html)
+          expect(rn).to eq(relnotes_from_url)
+        end
       end
     end
 
-    context "when a full language code is given (xx_XX)" do
-      it "returns product release notes for the given language" do
-        rn = reader.release_notes_for(product, user_lang: "en_US")
-        expect(rn.content).to eq("Release Notes\n")
-        expect(rn.lang).to eq("en_US")
-        expect(rn.user_lang).to eq("en_US")
-      end
+    context "when system is not registered" do
+      let(:registered?) { false }
 
-      context "and release notes are not available" do
-        it "returns product release notes for the short language code (xx)" do
-          rn = reader.release_notes_for(product, user_lang: "de_DE")
-          expect(rn.content).to eq("Versionshinweise\n")
-          expect(rn.lang).to eq("de")
-          expect(rn.user_lang).to eq("de_DE")
-        end
+      it "retrieves release notes from external sources" do
+        expect(url_reader).to receive(:release_notes).with(
+          Y2Packager::ReleaseNotesContentPrefs.new("cz_CZ", "en", :txt)
```

Comment:
JFYI: the language code is `cs`, the country code is `CZ`. Yes a bit confusing but the ISO language code uses that given language for the code, like for German it's `de`.... so the correct string here is `cs_CZ`.